### ISSUE TITLE
Creating a preset doesn't save across reboots

### ISSUE THREAD

**Issue by redfive2012** [crystalidea/macs-fan-control#389]
This has been affecting me for a while now, and I can't figure out why. I get all my fan speeds set based on temp values and try to save a preset. When I reboot the machine, the preset doesn't exist and I have to set it all up again and recreate the preset. It happens every time I reboot, but only in the Windows version. When I jump back into El Capitan, the preset I created ages ago is loaded and activated and everything works great.

**Comment by kleuter** [comment]
Windows 10?

**Comment by redfive2012** [comment]
Yes, Windows 10.

**Comment by kleuter** [comment]
I assume it happens when you DON'T quit the app manually. Can you please try to quit the app manually and then reboot? Thanks.

**Comment by redfive2012** [comment]
Ok. Sometimes I just reboot the computer, sometimes it hangs up and I just
turn it off and boot it again. I'll create a new profile, as I just booted
it up, and I'll quit and see what happens.

**Comment by redfive2012** [comment]
I saved a brand new preset called "Standard", then quit the program. I
started it right back up, and the preset was still there. I quit it again
and rebooted, and when it came back up the preset was not there.

Ever time I restart the program, whether I've booted the computer or I just
quit it and run it again, it has to pull up the UAC prompt. Could that have
something to do with it? Maybe it's saving the presets under some different
elevated privilege account (even though I'm the admin user), and the
account profile is reset on reboot?

**Comment by redfive2012** [comment]
My system seems to have rebooted itself last night, perhaps after applying
Windows updates. When I logged back in, it seems to have loaded the preset
that I had saved 4 days ago. This is the first time I've seen it keep the
preset available, let alone load it. I'll keep watch on it, and hopefully
it'll continue working with this preset.


*Nelson Butterworth*

**Comment by kleuter** [comment]
There's a bug but it's hard to catch. I basically need a 100% scenario of reproducing it in order to fix 😢

**Comment by redfive2012** [comment]
Indeed. I'm a developer myself, so I understand exactly what you mean, and
how frustrating it is to diagnose and solve an intermittent problem. I'm
willing to help you try to track it down.


*Nelson Butterworth*

**Comment by redfive2012** [comment]
Update: I recently had to reinstall Windows 10 from scratch (I replaced a mechanical HD with a SSD, but I kept getting errors when trying to clone). Since the reinstall, Macs Fan Control Windows has saved and loaded my profile just fine with every reboot. I've not had to recreate the profile, just saved it new once and updated it a few times after that. Any bugs were probably due to a problem with my old HDD, which did throw errors during the clone attempts (I'm thinking there were some bad blocks somewhere), rather than anything with the program itself.

**Comment by kleuter** [comment]
Thanks for the follow up, I'll keep this open for a while though.

**Comment by DaedalusEagle** [comment]
I am running Windows 10 and I have had a similar issue using Macs Fan Control. I have a 2020 27inch iMac. The issue seems to be sporadic. About 90% of the time when Windows loads my presets load just fine with it. However, on occasion my presets do not load and are removed completely. This is on a clean install of windows 10 using the latest version of the iso.

**Comment by kleuter** [comment]
Looks like the problem still happens for some users. 
Please.
Once the issue ever happens again (all presets are reset after new boot), please open regedit and show us what is under 
![image](https://user-images.githubusercontent.com/2600624/122544229-e7248200-d02c-11eb-8bce-d6fc71bd125d.png)

**Comment by kleuter** [comment]
Duplicate of #289

**Comment by JBreckeen** [comment]
Hello all.  I have the same issue as above.  MacPro 5.1 with Win10 only (no OSX).  Love Macs Fan (have Pro license) and works most of the time.  But every once in a while I happen to notice it is back in Automatic and my saved presets are gone.  It may or may not be after an update, or a reboot.

I can live with it losing the Presets sporatically, but really want to back up the presets so I can just put them back.  I have searched, but cannot find any sort of preferences file.  The question is:  how can I back up my PreSets?

Thanks in advance,

**Comment by kleuter** [comment]
For some users, also happen on macOS (iMac12,2, macOS 10.13)

**Comment by kleuter** [comment]
Related to https://github.com/crystalidea/macs-fan-control/issues/559
?

**Comment by redfive2012** [comment]
I updated mine over the weekend, and I also lost my preset.
